packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.test.tsx

```diff
@@ -0,0 +1,304 @@
+          expect(response.isItemExpanded('2')).to.equal(false);
+        }).toErrorDev(
+          'MUI X: A component is changing the default expandedItems state of an uncontrolled TreeView after being initialized. To suppress this warning opt to use a controlled TreeView.',
+        );
+      });
+    });
+
+    describe('click interactions', () => {
+      it('should expand collapsed item when clicking on an item content', () => {
+        const response = render({
+          items: [{ id: '1', children: [{ id: '1.1' }] }, { id: '2' }],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(false);
+        fireEvent.click(response.getItemContent('1'));
+        expect(response.isItemExpanded('1')).to.equal(true);
+      });
+
+      it('should collapse expanded item when clicking on an item content', () => {
+        const response = render({
+          items: [{ id: '1', children: [{ id: '1.1' }] }, { id: '2' }],
+          defaultExpandedItems: ['1'],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(true);
+        fireEvent.click(response.getItemContent('1'));
+        expect(response.isItemExpanded('1')).to.equal(false);
+      });
+
+      it('should not expand collapsed item when clicking on a disabled item content', () => {
+        const response = render({
+          items: [{ id: '1', disabled: true, children: [{ id: '1.1' }] }, { id: '2' }],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(false);
+        fireEvent.click(response.getItemContent('1'));
+        expect(response.isItemExpanded('1')).to.equal(false);
+      });
+
+      it('should not collapse expanded item when clicking on a disabled item', () => {
+        const response = render({
+          items: [{ id: '1', disabled: true, children: [{ id: '1.1' }] }, { id: '2' }],
+          defaultExpandedItems: ['1'],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(true);
+        fireEvent.click(response.getItemContent('1'));
+        expect(response.isItemExpanded('1')).to.equal(true);
+      });
+
+      it('should expand collapsed item when clicking on an item label', () => {
+        const response = render({
+          items: [{ id: '1', children: [{ id: '1.1' }] }, { id: '2' }],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(false);
+        fireEvent.click(response.getItemLabel('1'));
+        expect(response.isItemExpanded('1')).to.equal(true);
+      });
+
+      it('should expand collapsed item when clicking on an item icon container', () => {
+        const response = render({
+          items: [{ id: '1', children: [{ id: '1.1' }] }, { id: '2' }],
+        });
+
+        expect(response.isItemExpanded('1')).to.equal(false);
+        fireEvent.click(response.getItemIconContainer('1'));
+        expect(response.isItemExpanded('1')).to.equal(true);
+      });
+
+      it('should be able to limit the expansion to the icon', function test() {
```

Comment:
New test
This customization is simple enough and common enough that IMHO it's worth testing